Plugins/NativeCamera/Source/NativeCamera.cpp

```diff
@@ -0,0 +1,166 @@
+
+            video.Set("videoWidth", Napi::Number::New(env, maxWidth).As<Napi::Value>());
+            video.Set("videoHeight", Napi::Number::New(env, maxHeight).As<Napi::Value>());
+            video.Set("frontCamera", Napi::Boolean::New(env, (facingMode == "user")).As<Napi::Value>());
+            video.Set("readyState", Napi::Number::New(env, 10).As<Napi::Value>());
+            video.Set("HAVE_CURRENT_DATA", Napi::Number::New(env, 1).As<Napi::Value>());
+            video.Set("isNative", Napi::Boolean::New(env, true).As<Napi::Value>());
+
+            video.Set("setAttribute", Napi::Function::New(env,[](const Napi::CallbackInfo& /*info*/){
+                }));
+
+            video.Set("addEventListener", Napi::Function::New(env, [](const Napi::CallbackInfo& info) {
+                auto callback = info[1].As<Napi::Function>();
+                info.This().ToObject().Set("_cb", callback.As<Napi::Object>().As<Napi::Value>());
+                }));
+
+            video.Set("removeEventListener", Napi::Function::New(env, [](const Napi::CallbackInfo& ) { }));
+
+            video.Set("play", Napi::Function::New(env, [](const Napi::CallbackInfo& info) {
+                auto _this = info.This().ToObject();
+                auto cb = _this.Get("_cb");
+                if (!cb.IsFunction())
+                {
+                    throw Napi::Error::New(info.Env(), "Invalid function for video playback.");
+                }
+                cb.As<Napi::Function>().Call({ });
+                }));
+
+            video.Set("pause", Napi::Function::New(env, [](const Napi::CallbackInfo& info) {
+                    auto _this = info.This().ToObject();
+                    auto cameraDataObject = _this.Get(JS_NATIVECAMERA_DATA);
+                    if (cameraDataObject.IsExternal())
+                    {
+                        const auto* cameraInterface = cameraDataObject.As<Napi::External<CameraInterface>>().Data();
+                        delete cameraInterface;
+                        _this.Set(JS_NATIVECAMERA_DATA, info.Env().Undefined());
+                    }
+                }));
+
+            return video;
+        }
+
+        void UpdateVideoTexture(const Napi::CallbackInfo& info)
+        {
+            const auto texture = info[0].As<Napi::External<TextureData>>().Data();
+            auto videoObject = info[1].As<Napi::Object>();
+
+            auto cameraInterfaceObject = videoObject.Get(JS_NATIVECAMERA_DATA);
+            if (!cameraInterfaceObject.IsExternal())
+            {
+                const auto width = videoObject.Get("videoWidth").As<Napi::Number>().Uint32Value();
+                const auto height = videoObject.Get("videoHeight").As<Napi::Number>().Uint32Value();
+                const bool frontCamera = videoObject.Get("frontCamera").As<Napi::Boolean>().Value();
+
+                cameraInterfaceObject = Napi::External<CameraInterface>::New(info.Env(), CameraInterface::CreateInterface(width, height, frontCamera));
```

Comment:
The lifecycle of this thing is kind of confusing. If it's destroyed on pause, should it be created on play?

Reply:
yes, might be less confusing.